var/spack/repos/builtin/packages/fftw/package.py

```diff
@@ -73,8 +73,32 @@ class Fftw(AutotoolsPackage):
 
     @property
     def libs(self):
-        result = find_libraries(['libfftw3'], root=self.prefix, recurse=True)
-        return result
+
+        query_parameters = self.spec.last_query.extra_parameters
+
+        # List of all the suffixes associated with float precisions
+        precisions = [
+            ('float', 'f'),
+            ('double', ''),
+            ('long_double', 'l'),
+            ('quad', 'q')
+        ]
+        query_parameters.append('double')
+
+        # Retrieve the correct suffix and make double the default
+        sfx = next(v for k, v in precisions if k in query_parameters)
```

Comment:
@alalazo Since only a single query parameter is used, IMO it would be clearer to say something like 

```
chosen_precision = query_parameters[0] if query_parameters else 'double'
```

Also it appears that if `long_double` is in `query_parameters` that `double` will actually be chosen instead (since you are iterating through `precisions` instead of through `query_parameters` and `double` comes before `long_double`). In other words it appears that `long_double` or `quad` would never be used.

Reply:
> Since only a single query parameter is used, IMO it would be clearer to say something like

I don't use a single query parameter (see `mpi` and `openmp` a few lines below). Also:
```
chosen_precision = query_parameters[0] if query_parameters else 'double'
```
enforces a positional order (i.e. precision must be there and must be first), while I want to serve in the same way things like:
```
spec['fftw:mpi'].libs
spec['fftw:mpi,double'].libs
spec['fftw:double,mpi'].libs
```
The only part I was dubious here is if I should treat multiple precision tags simultaneously.

> Also it appears that if long_double is in query_parameters that double will actually be chosen instead (since you are iterating through precisions instead of through query_parameters and double comes before long_double). In other words it appears that long_double or quad would never be used.

`query_parameters` is a list-like object, so I don't really get this comment:
```console
>>> precisions = [
...             ('float', 'f'),
...             ('double', ''),
...             ('long_double', 'l'),
...             ('quad', 'q')
...         ]
>>> query_parameters = ['long_double', 'mpi']
>>> next(k for k, _ in precisions if k in query_parameters)
'long_double'
```
In case of multiple precision tags (which should not be there) the behavior is to prefer `float`, over `double` over `long_double` over `quad`:
```console
>>> query_parameters = ['openmp', 'long_double','float']
>>> next(k for k, _ in precisions if k in query_parameters)
'float'
```
Do you want to change this default or be more defensive?

Reply:
@alalazo It looks like you are always appending `'double'` to `query_parameters` so I thought the example would be something like:

```
>>> query_parameters = ['openmp','quad','double'] # the last 'double' was added automatically on line 86
>>> next(k for k, _ in precisions if k in query_parameters)
'double'
```

(to be clear that's what I assume would happen. If it works for you then sorry about all the fuss)

Reply:
@scheibelp Ah ok, that makes sense. I definitely overlooked appending. I'll improve this and ping you for another look at it.